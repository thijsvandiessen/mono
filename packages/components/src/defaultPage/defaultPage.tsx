import React from 'react'
import { PageContent } from '../pageContent'
import { getPage } from '@crea/graphql/src/getters/getPage'
import { notFound } from 'next/navigation'

export interface Props {
  slug: string
}

export const DefaultPage = async ({ slug }: Props) => {
  const { data } = await getPage({ slug })
  if (!data) return notFound()

  return (
    <article>
      <PageContent
        sectionClassName="content-layout"
        items={data.content}
        pageTitle={data.title || ''}
      />
    </article>
  )
}
