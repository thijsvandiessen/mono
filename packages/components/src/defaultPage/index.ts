export { DefaultPage } from './defaultPage.jsx'
