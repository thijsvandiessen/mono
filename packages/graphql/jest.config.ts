import { type Config } from 'jest'

const config: Config = {
  // Add more setup options before each test is run
  setupFilesAfterEnv: ['./jest.setup.cjs'],
  testEnvironment: 'jest-environment-jsdom',
  preset: 'ts-jest',
  transform: {
    '^.+\\.tsx?$': [
      'ts-jest',
      {
        babel: true,
        tsconfig: 'tsconfig.test.json',
      },
    ],
  },
  collectCoverage: true,
  coverageDirectory: 'coverage',
}

export default config
