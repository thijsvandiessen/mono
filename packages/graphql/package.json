{
  "name": "@mono/graphql",
  "version": "1.0.0",
  "description": "",
  "type": "module",
  "exports": {
    ".": {
      "import": "./src/index.ts",
      "module": "./dist/graphql.js",
      "types": "./dist/graphql.d.ts"
    }
  },
  "main": "./dist/graphql.js",
  "types": "./dist/graphql.d.ts",
  "files": [
    "./dist/*"
  ],
  "scripts": {
    "build": "vite build",
    "codegen": "graphql-codegen --config ./codegen.ts",
    "codegen:watch": "graphql-codegen --require dotenv/config --config ./codegen.ts --watch",
    "download-schema": "rover graph fetch vandiessen-mono-graph@current > ./src/generated/schema.graphql && npm run lint:styles",
    "lint": "npm run lint:js && npm run lint:styles && npm run lint:typescript",
    "lint:js": "eslint . --fix",
    "lint:styles": "prettier . --write --cache --log-level silent --list-different",
    "lint:typescript": "tsc",
    "test": "jest"
  },
  "dependencies": {
    "@mono/utils": "1.0.0",
    "@urql/core": "5.0.6",
    "@urql/exchange-graphcache": "7.1.2",
    "@urql/exchange-retry": "1.3.0",
    "graphql": "16.9.0",
    "react": "18.3.1"
  },
  "devDependencies": {
    "@apollo/rover": "0.26.2",
    "@commitlint/config-conventional": "19.2.2",
    "@graphql-codegen/cli": "5.0.2",
    "@graphql-codegen/client-preset": "4.3.3",
    "@graphql-codegen/introspection": "4.0.3",
    "@graphql-codegen/typescript": "4.0.9",
    "@graphql-codegen/typescript-document-nodes": "4.0.9",
    "@graphql-codegen/typescript-operations": "4.2.3",
    "@testing-library/jest-dom": "6.5.0",
    "@testing-library/react": "16.0.1",
    "@types/jest": "29.5.12",
    "@types/node": "22.5.4",
    "@types/react": "18.3.5",
    "@types/react-dom": "18.2.18",
    "@typescript-eslint/eslint-plugin": "8.5.0",
    "@typescript-eslint/parser": "8.5.0",
    "dotenv": "16.4.5",
    "eslint": "8.57.0",
    "eslint-config-next": "14.2.8",
    "eslint-plugin-react-hooks": "4.6.2",
    "eslint-plugin-testing-library": "6.3.0",
    "jest": "29.7.0",
    "jest-environment-jsdom": "29.7.0",
    "prettier": "3.3.3",
    "pretty-quick": "4.0.0",
    "turbo": "2.1.1",
    "typescript": "5.5.4"
  }
}
