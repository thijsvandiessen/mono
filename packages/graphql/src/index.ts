export { defaultExchange } from './cacheExchange'
export { client } from './gqlClient'
export { linkConfig } from './linkConfig'
export {
  getEvent,
  getEventPage,
  getEventPageSeo,
  getEvents,
  getEventsMeta,
  getEventUrls,
  getFutureEvents,
  getGeneralInfo,
  getLocation,
  getLocations,
  getPage,
  getPages,
  getPageSeo,
  getPagesMeta,
  getSiteInfo,
  getSiteMetadata,
} from './getters'

export type {
  Author,
  BlogPost,
  Event,
  PageLink,
  CloudinaryAsset,
  Image,
  Location,
  Music,
} from './types'

export { isOfTypeCloudinaryAsset } from './types'
export { formatCloudinaryDocument, formatCloudinaryImage } from './formatters'
export * from './generated/graphql'
