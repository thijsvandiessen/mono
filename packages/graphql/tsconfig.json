{
  "extends": "../../tsconfig.json",
  "compilerOptions": {
    "lib": ["ESNext", "DOM"]
  },
  "include": ["src/**/*", "src/*"],
  "exclude": ["src/**/*.test.ts", "src/**/*.test.tsx"]
}
