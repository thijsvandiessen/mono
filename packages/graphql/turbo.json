{
  "extends": ["//"],
  "pipeline": {
    "lint": {},
    "test": {},
    "lint:js": {},
    "lint:styles": {},
    "lint:typescript": {}
  },
  "remoteCache": {
    "signature": false
  },
  "globalDependencies": ["tsconfig.json"]
}
