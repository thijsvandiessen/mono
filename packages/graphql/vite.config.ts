import { defineConfig } from 'vite'
import dts from 'vite-plugin-dts'

export default defineConfig({
  build: {
    lib: {
      entry: ['./src/index.ts'],
      name: 'graphql',
      fileName: 'graphql',
      formats: ['es'],
    },
    rollupOptions: {
      output: {
        inlineDynamicImports: false,
        preserveModules: true,
        dir: 'dist',
      },
    },
    sourcemap: true,
  },

  plugins: [dts({ outDir: 'dist', rollupTypes: true })],
})
