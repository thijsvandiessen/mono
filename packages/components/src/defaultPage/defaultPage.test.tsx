import React from 'react'
import { DefaultPage } from './defaultPage'
import type { Props } from './defaultPage'
import { getPage } from '@mono/graphql'
import { resolvedComponent } from '@mono/utils'
import { render } from '@testing-library/react'

jest.mock('../navigation', () => {
  const originalModule = jest.requireActual('../navigation')
  return {
    __esModule: true,
    ...originalModule,
    Navigation: jest.fn(() => <div>[Navigation]</div>),
  }
})

jest.mock('@mono/graphql', () => {
  const originalModule = jest.requireActual('@mono/graphql')
  return {
    __esModule: true,
    ...originalModule,
    getPage: jest.fn(),
  }
})

const getPageMock = jest.mocked(getPage)

describe('Concert component', () => {
  it('shows all the data', async () => {
    getPageMock.mockResolvedValue({
      data: {
        __typename: 'PageRecord',
        title: 'title',
        slug: 'contact',
        _createdAt: 'some date time string',
        _updatedAt: 'some date time string',
        id: 'unique-id',
        content: [],
      },
      error: undefined,
    })

    const Resolved = await resolvedComponent<Props>(DefaultPage, {
      slug: 'contact',
    })

    const { container } = render(<Resolved />)

    expect(container).toMatchSnapshot()
  })
})
