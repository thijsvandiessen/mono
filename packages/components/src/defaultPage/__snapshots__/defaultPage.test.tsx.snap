// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Concert component shows all the data 1`] = `
<div>
  <article />
</div>
`;
